cmd/juju/model/defaultscommand.go

```diff
@@ -214,8 +217,8 @@ func (c *defaultsCommand) Init(args []string) error {
 		// set args.
 		return c.handleSetArgs(args)
 	case len(args) == 0:
-		// This should be reset only (but possibly for a region).
-		return c.handleZeroArgs()
+		c.action = c.getDefaults
```

Comment:
This change causes `juju model-defaults --reset foo` to reset foo and print out all the model defaults. Also in the case that one names a cloud/region it resets the key(s) and prints out the args that have a setting in that region.